src/workflows/integration/scgpt_leiden/main.nf

```diff
@@ -0,0 +1,169 @@
+    )
+    | cross_check_genes.run(
+      fromState: { id, state ->
+      // Check whether the genes are part of the provided vocabulary. Subsets for genes present in vocab only.
+        [
+          "input": state.input,
+          "modality": state.modality,
+          "vocab_file": state.model_vocab,
+          "var_gene_names": state.var_gene_names,
+          "output": state.output,
+          "pad_token": state.pad_token
+        ]
+      },
+      toState: ["input": "output"]
+    )
+    | binning.run(
+      // Bins the data into a fixed number of bins.
+        fromState: {id, state -> [
+            "input": state.input,
+            "modality": state.modality,
+            "input_layer": state.input_layer,
+            "n_input_bins": state.n_input_bins,
+            "binned_layer": "binned",
+            "output": state.output
+          ]
+        },
+        toState: ["input": "output"]
+    )
+    | pad_tokenize.run(
+      // Padding and tokenization of gene count values.
+       fromState: {id, state -> [
+            "input": state.input,
+            "modality": state.modality,
+            "model_vocab": state.model_vocab,
+            "input_layer": "binned",
+            "var_gene_names": state.var_gene_names,
+            "pad_token": state.pad_token,
+            "pad_value": state.pad_value,
+            "max_seq_len": state.max_seq_len,
+            "obsm_gene_tokens": "gene_id_tokens",
+            "obsm_tokenized_values": "values_tokenized",
+            "obsm_padding_mask": "padding_mask",
+            "output": state.output
+          ]
+        },
+        toState: ["input": "output"]
+    )
+    | embedding.run(
+      // Generation of cell embedings from the tokenized gene counts values.
+      fromState: {id, state -> [
+          "input": state.input,
+          "modality": state.modality,
+          "model": state.model,
+          "model_vocab": state.model_vocab,
+          "model_config": state.model_config,
+          "obsm_gene_tokens": "gene_id_tokens",
+          "obsm_tokenized_values": "values_tokenized",
+          "obsm_padding_mask": "padding_mask",
+          "var_gene_names": state.var_gene_names,
+          "obs_batch_label": state.obs_batch_label,
+          "pad_token": state.pad_token,
+          "pad_value": state.pad_value,
+          "DSBN": state.DSBN,
+          "batch_size": state.batch_size,
+          "obsm_embeddings": "X_scGPT",
+          "output": state.output
+        ]
+      },
+      toState: ["input": "output"]
+    )
+
+    | find_neighbors.run(
+      fromState: {id, state -> [
+          "input": state.input,
+          "uns_output": "scGPT_integration_neighbors",
+          "obsp_distances": "scGPT_integration_distances",
+          "obsp_connectivities": "scGPT_integration_connectivities",
+          "obsm_input": "X_scGPT",
+          "modality": state.modality
+        ]
+      },
+      toState: ["input": "output"]
+    )
+
+with_leiden_ch = neighbors_ch
```

Comment:
You can avoid using two channels here by using the `if` argument for `.run` with `id, state -> state.leiden_resolution`  (you will need it both for `leiden` and `move_obsm_to_obs`)

Reply:
oh neat